apps/api-extractor/src/api/Extractor.ts

```diff
@@ -434,12 +358,134 @@ export class Extractor {
       compilerState,
       extractorConfig,
       succeeded,
-      apiReportChanged,
+      apiReportChanged: anyReportChanged,
       errorCount: messageRouter.errorCount,
       warningCount: messageRouter.warningCount
     });
   }
 
+  /**
+   * Generates the API report at the specified release level, writes it to the specified file path, and compares
+   * the output to the existing report (if one exists).
+   *
+   * @param reportTempDirectoryPath - The path to the directory under which the temp report file will be written prior
+   * to comparison with an existing report.
+   * @param reportDirectoryPath - The path to the directory under which the existing report file is located, and to
+   * which the new report will be written post-comparison.
+   * @param reportVariant - Determines which API members will be included in the report, based on their tagged release levels.
+   *
+   * @returns Whether or not the newly generated report differs from the existing report (if one exists).
+   */
+  private static _writeApiReport(
+    collector: Collector,
+    extractorConfig: ExtractorConfig,
+    messageRouter: MessageRouter,
+    reportTempDirectoryPath: string,
+    reportDirectoryPath: string,
+    reportFileName: string,
+    reportVariant: ApiReportVariant,
+    localBuild: boolean
+  ): boolean {
+    let apiReportChanged: boolean = false;
```

Comment:
This logic was moved mostly as is. The only changes made are with regards to how `reportVariant` is leveraged.